### PR TITLE
Feature/nijo 1129 manual booking manage flow

### PR DESCRIPTION
**Pull request by mingazhev**
1. Created a separate BookingManagementService to be used by admins, agents and service account
2. Created proxy servers to manage bookings for agents and admins
3. Added discard logic, available only from admin endpoints

### CODE REVIEW

File: Api/Services/Accommodations/Bookings/Management/BookingManagementService.cs [hunk 1]
```diff
@@ -118,47 +48,78 @@ public class BookingManagementService : IBookingManagementService
...
+            {
+                _logger.LogBookingRefreshStatusFailure($"Failed to refresh status for a booking with reference code: '{referenceCode}' " +
+                    $"while getting info from a supplier. Error: {getDetailsError}");
+                
+                return Result.Failure(getDetailsError.Detail);
+            }
+
+            await _responseProcessor.ProcessResponse(newDetails);
+
+            _logger.LogBookingRefreshStatusSuccess($"Successfully refreshed status fot a booking with reference code: '{referenceCode}'. " +
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/Management/BookingManagementService.cs:108]: fo**t** a booking

File: Api/Services/Accommodations/Bookings/ResponseProcessing/IBookingChangesProcessor.cs [hunk 2]
```diff
@@ -14,5 +14,7 @@ public interface IBookingChangesProcessor
         Task ProcessBookingNotFound(Booking booking, EdoContracts.Accommodations.Booking bookingResponse);
 
         Task ProcessRejection(Booking booking, UserInfo internalServiceAccount);
+
+        Task<Result> ProcessDiscarding(Booking booking, UserInfo user);
```
Comment (kirillta) [inline Api/Services/Accommodations/Bookings/ResponseProcessing/IBookingChangesProcessor.cs:18]: `ProcessDiscarding` equals `Discard` isn't it?

File: Api/Controllers/AdministratorControllers/BookingsController.cs [hunk 3]
```diff
@@ -1,42 +1,58 @@
...
+        public BookingsController(IAdministratorContext administratorContext,
+            IAdministratorBookingManagementService bookingManagementService)
         {
             _administratorContext = administratorContext;
             _bookingManagementService = bookingManagementService;
         }
+        
+        
+        /// <summary>
+        ///     Cancel accommodation booking by admin.
```
Comment (kirillta) [inline Api/Controllers/AdministratorControllers/BookingsController.cs:27]: Cancel**s**

File: Api/AdministratorServices/AdministratorBookingManagementService.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+using HappyTravel.Edo.Api.Models.Users;
+using HappyTravel.Edo.Api.Services.Accommodations.Bookings.Management;
+using HappyTravel.Edo.Data.Bookings;
+using HappyTravel.Edo.Data.Management;
+
+namespace HappyTravel.Edo.Api.AdministratorServices
+{
+    public class AdministratorBookingManagementService : IAdministratorBookingManagementService
+    {
+        public AdministratorBookingManagementService(IBookingRecordsManager recordsManager,
```
Comment (kirillta) [inline Api/AdministratorServices/AdministratorBookingManagementService.cs:12]: IBookingRecor**dM**anager

**Review by kirillta — APPROVED** [review]

File: Api/Services/Accommodations/Bookings/Management/BookingManagementService.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/Management/BookingManagementService.cs:108]: Fixed

File: Api/Controllers/AdministratorControllers/BookingsController.cs (hunk 3 above)
Comment (mingazhev) [reply Api/Controllers/AdministratorControllers/BookingsController.cs:27]: fixed

File: Api/AdministratorServices/AdministratorBookingManagementService.cs (hunk 4 above)
Comment (mingazhev) [reply Api/AdministratorServices/AdministratorBookingManagementService.cs:12]: fixed

File: Api/Services/Accommodations/Bookings/ResponseProcessing/IBookingChangesProcessor.cs (hunk 2 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Bookings/ResponseProcessing/IBookingChangesProcessor.cs]: Fixed with refactoring

**Pull request merged**